src/service.js

```diff
@@ -318,7 +318,16 @@ export function getParentWindow(win) {
  * @return {!Window}
  */
 export function getTopWindow(win) {
-  return win.__AMP_TOP || win;
+  const top = win.__AMP_TOP;
+  if (top === null) {
+    return win;
+  }
+  if (top != null) {
+    return top;
+  }
+  // Make this defined but null to speed up future lookups.
+  win.__AMP_TOP = null;
```

Comment:
We could set this to `win` instead? Then keep the same `||`: `win.__AMP_TOP || (win.__AMP_TOP = win)`